src/metabase/api/common.clj

```diff
@@ -713,3 +713,17 @@
   (cond-> item
     (:archived_directly item)
     (assoc :collection_id trash-collection-id)))
+
+(defn present-items
+  "A convenience function that takes a heterogeneous collection of items. Each item should have, at minimum, a `:model`
+  and an `:id`. The `f` function is called like `(f model all-items-with-that-model)` and should return a collection
```

Comment:
Should this be enforced on items with mu/defn?

Reply:
Sounds good, I'll add this.